lib/truffle/truffle/psych.rb

```diff
@@ -17,30 +17,95 @@ def self.libyaml_version
     [1, 14, 0]
   end
 
-
   class ClassLoader
-
     def path2class(path)
       eval("::#{path}")
     end
-
   end
 
   module Visitors
     class ToRuby
-
       def build_exception(klass, mesg)
         klass.new(mesg)
       end
-
     end
 
     class YAMLTree
-
       def private_iv_get(target, prop)
         target.instance_variable_get("@#{prop}")
       end
+    end
+  end
+
+  class Parser
+    def parse(yaml, path = nil)
+      tainted = yaml.tainted? || yaml.is_a?(IO)
+      if !(String === yaml) && yaml.respond_to?(:read)
+        # TODO (eregon, 19 Jan. 2018): is it worth using a streaming adapter?
+        yaml = yaml.read
+      else
+        yaml = StringValue(yaml)
+      end
 
+      if path.nil? && yaml.respond_to?(:path)
+        path = yaml.path
+      end
+
+      parser = Truffle.invoke_primitive(:pysch_create_parser, yaml)
+
+      loop do
+        begin
+          event = get_event(parser)
+        rescue RubyTruffleError => e
+          if info = parse_exception_info(e)
+            raise Psych::SyntaxError.new(path, *info)
+          else
+            raise e
+          end
+        end
+
+        if event?(event, :StreamStart)
+          @handler.start_stream(ANY)
+        elsif event?(event, :DocumentStart)
+          tags = []
+          version, explicit = doc_start_info(event) { |key, value|
```

Comment:
I think we've settled on using `do...end` for multi-line blocks.

Reply:
It looks like there's a cop for this: http://rubocop.readthedocs.io/en/latest/cops_style/#styleblockdelimiters

I can do up a PR to enable that and clean up whatever code we have violating the style currently.

Reply:
:+1: I think that would help to keep it consistent.